src/pybind/mgr/cephadm/module.py

```diff
@@ -1102,27 +1130,30 @@ def _run_cephadm(self,
 
             self.log.debug('args: %s' % (' '.join(final_args)))
             if self.mode == 'root':
-                if stdin:
-                    self.log.debug('stdin: %s' % stdin)
-                script = 'injected_argv = ' + json.dumps(final_args) + '\n'
-                if stdin:
-                    script += 'injected_stdin = ' + json.dumps(stdin) + '\n'
-                script += self._cephadm
-                python = connr.choose_python()
-                if not python:
-                    raise RuntimeError(
-                        'unable to find python on %s (tried %s in %s)' % (
-                            host, remotes.PYTHONS, remotes.PATH))
                 try:
+                    python = connr.choose_python()
+
+                    mgrd = self.cache.get_daemon("mgr.%s" % self.get_mgr_id())
+                    cephadm_dest = '/var/lib/ceph/%s/cephadm/%s/%s/cephadm' % (
+                        self._cluster_fsid, self.get_mgr_id(), mgrd.container_image_id)
```

Comment:
this looks already good. There is just one problem: in a vstart cluster, there is no 
containerized MGR. Therefore I think we can simplify this to:

```suggestion
                    cephadm_hash = sha256(self._read_cephadm_script())
                    cephadm_dest = f'/var/lib/ceph/{self._cluster_fsid}/cephadm-{cephadm_hash}/cephadm'
```

Reply:
ping. Needs rebase: now you also need to fix

https://github.com/ceph/ceph/blob/fd1c8dd561b7e24bc400ace48dd9edaefa9f0ec4/src/pybind/mgr/cephadm/module.py#L1930-L1939

to contain the hash.